File: yahtzee/jdantonio/test/simple_sum_rules_test.js
var yahtzee = require('../lib/yahtzee.js');

exports.testOnesRuleWithZeroOnes = function(test) {
  var expected = 0;
  var actual = yahtzee.score('Ones', [2, 3, 4, 5, 6]);
  test.equal(expected, actual);
  test.done();
};

exports.testOnesRuleWithSomeOnes = function(test) {
  var expected = 3;
  var actual = yahtzee.score('Ones', [1, 2, 1, 4, 1]);
  test.equal(expected, actual);
  test.done();
};

exports.testOnesRuleWithAllOnes = function(test) {
  var expected = 5;
  var actual = yahtzee.score('Ones', [1, 1, 1, 1, 1]);
  test.equal(expected, actual);
  test.done();
};

exports.testTwosRuleWithZeroTwos = function(test) {
  var expected = 0;
  var actual = yahtzee.score('Twos', [1, 3, 4, 5, 6]);
  test.equal(expected, actual);
  test.done();
};

exports.testTwosRuleWithSomeTwos = function(test) {
  var expected = 4;
  var actual = yahtzee.score('Twos', [1, 2, 3, 4, 2]);
  test.equal(expected, actual);
  test.done();
};

exports.testTwosRuleWithAllTwos = function(test) {
  var expected = 10;
  var actual = yahtzee.score('Twos', [2, 2, 2, 2, 2]);
  test.equal(expected, actual);
  test.done();
};

exports.testThreesRuleWithZeroThrees = function(test) {
  var expected = 0;
  var actual = yahtzee.score('Threes', [1, 2, 4, 5, 6]);
  test.equal(expected, actual);
  test.done();
};

exports.testThreesRuleWithSomeThrees = function(test) {
  var expected = 6;
  var actual = yahtzee.score('Threes', [1, 3, 1, 3, 1]);
  test.equal(expected, actual);
  test.done();
};

exports.testThreesRuleWithAllThrees = function(test) {
  var expected = 15;
  var actual = yahtzee.score('Threes', [3, 3, 3, 3, 3]);
  test.equal(expected, actual);
  test.done();
};

exports.testFoursRuleWithZeroFours = function(test) {
  var expected = 0;
  var actual = yahtzee.score('Fours', [1, 2, 3, 5, 6]);
  test.equal(expected, actual);
  test.done();
};

exports.testFoursRuleWithSomeFours = function(test) {
  var expected = 12;
  var actual = yahtzee.score('Fours', [4, 1, 4, 1, 4]);
  test.equal(expected, actual);
  test.done();
};

exports.testFoursRuleWithAllFours = function(test) {
  var expected = 20;
  var actual = yahtzee.score('Fours', [4, 4, 4, 4, 4]);
  test.equal(expected, actual);
  test.done();
};

exports.testFivesRuleWithZeroFives = function(test) {
  var expected = 0;
  var actual = yahtzee.score('Fives', [1, 2, 3, 4, 6]);
  test.equal(expected, actual);
  test.done();
};

exports.testFivesRuleWithSomeFives = function(test) {
  var expected = 10;
  var actual = yahtzee.score('Fives', [1, 1, 1, 5, 5]);
  test.equal(expected, actual);
  test.done();
};

exports.testFivesRuleWithAllFives = function(test) {
  var expected = 25;
  var actual = yahtzee.score('Fives', [5, 5, 5, 5, 5]);
  test.equal(expected, actual);
  test.done();
};

exports.testSixesRuleWithZeroSixes = function(test) {
  var expected = 0;
  var actual = yahtzee.score('Sixes', [1, 2, 3, 4, 5]);
  test.equal(expected, actual);
  test.done();
};

exports.testSixesRuleWithSomeSixes = function(test) {
  var expected = 24;
  var actual = yahtzee.score('Sixes', [1, 6, 6, 6, 6]);
  test.equal(expected, actual);
  test.done();
};

exports.testSixesRuleWithAllSixes = function(test) {
  var expected = 30;
  var actual = yahtzee.score('Sixes', [6, 6, 6, 6, 6]);
  test.equal(expected, actual);
  test.done();
};

File: yahtzee/jdantonio/golang/simple_sum_rules_test.go
package yahtzee

import (
	"testing"
)

func TestOnesRuleWithZeroOnes(t *testing.T) {
	expected := 0
	actual := Score(Ones, [5]int{2, 3, 4, 5, 6})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestOnesRuleWithSomeOnes(t *testing.T) {
	expected := 3
	actual := Score(Ones, [5]int{1, 2, 1, 4, 1})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestOnesRuleWithAllOnes(t *testing.T) {
	expected := 5
	actual := Score(Ones, [5]int{1, 1, 1, 1, 1})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestTwosRuleWithZeroTwos(t *testing.T) {
	expected := 0
	actual := Score(Twos, [5]int{1, 3, 4, 5, 6})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestTwosRuleWithSomeTwos(t *testing.T) {
	expected := 4
	actual := Score(Twos, [5]int{1, 2, 3, 4, 2})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestTwosRuleWithAllTwos(t *testing.T) {
	expected := 10
	actual := Score(Twos, [5]int{2, 2, 2, 2, 2})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestThreesRuleWithZeroThrees(t *testing.T) {
	expected := 0
	actual := Score(Threes, [5]int{1, 2, 4, 5, 6})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestThreesRuleWithSomeThrees(t *testing.T) {
	expected := 6
	actual := Score(Threes, [5]int{1, 3, 1, 3, 1})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestThreesRuleWithAllThrees(t *testing.T) {
	expected := 15
	actual := Score(Threes, [5]int{3, 3, 3, 3, 3})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestFoursRuleWithZeroFours(t *testing.T) {
	expected := 0
	actual := Score(Fours, [5]int{1, 2, 3, 5, 6})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestFoursRuleWithSomeFours(t *testing.T) {
	expected := 12
	actual := Score(Fours, [5]int{4, 1, 4, 1, 4})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestFoursRuleWithAllFours(t *testing.T) {
	expected := 20
	actual := Score(Fours, [5]int{4, 4, 4, 4, 4})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestFivesRuleWithZeroFives(t *testing.T) {
	expected := 0
	actual := Score(Fives, [5]int{1, 2, 3, 4, 6})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestFivesRuleWithSomeFives(t *testing.T) {
	expected := 10
	actual := Score(Fives, [5]int{1, 1, 1, 5, 5})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestFivesRuleWithAllFives(t *testing.T) {
	expected := 25
	actual := Score(Fives, [5]int{5, 5, 5, 5, 5})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestSixesRuleWithZeroSixes(t *testing.T) {
	expected := 0
	actual := Score(Sixes, [5]int{1, 2, 3, 4, 5})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestSixesRuleWithSomeSixes(t *testing.T) {
	expected := 24
	actual := Score(Sixes, [5]int{1, 6, 6, 6, 6})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}

func TestSixesRuleWithAllSixes(t *testing.T) {
	expected := 30
	actual := Score(Sixes, [5]int{6, 6, 6, 6, 6})
	if actual != expected {
		t.Errorf("expected %v got %v", expected, actual)
	}
}
